From 2bd6855d2e6a570ee3b68cbe46a2ab714e5dda0a Mon Sep 17 00:00:00 2001
From: Martin Whitaker <fsf@martin-whitaker.me.uk>
Date: Mon, 25 May 2020 21:02:10 +0200
Subject: grub-install: Fix inverted test for NLS enabled when copying locales

Commit 3d8439da8 (grub-install: Locale depends on nls) attempted to avoid
copying locale files to the target directory when NLS was disabled.
However the test is inverted, and it does the opposite.

Signed-off-by: Martin Whitaker <fsf@martin-whitaker.me.uk>
Signed-off-by: Daniel Kiper <daniel.kiper@oracle.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>

Origin: upstream, https://git.savannah.gnu.org/cgit/grub.git/commit/?id=8451454889f3b46312cc16eb07970d6c20da0812
Last-Update: 2021-02-07

Patch-Name: grub-install-inverted-nls-test.patch
---
 util/grub-install-common.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/util/grub-install-common.c b/util/grub-install-common.c
index 61f9075bc..1d47c33c0 100644
--- a/util/grub-install-common.c
+++ b/util/grub-install-common.c
@@ -673,7 +673,7 @@ copy_all (const char *srcd,
   grub_util_fd_closedir (d);
 }
 
-#if !(defined (GRUB_UTIL) && defined(ENABLE_NLS) && ENABLE_NLS)
+#if (defined (GRUB_UTIL) && defined(ENABLE_NLS) && ENABLE_NLS)
 static const char *
 get_localedir (void)
 {
@@ -743,7 +743,7 @@ static void
 grub_install_copy_nls(const char *src __attribute__ ((unused)),
                       const char *dst __attribute__ ((unused)))
 {
-#if !(defined (GRUB_UTIL) && defined(ENABLE_NLS) && ENABLE_NLS)
+#if (defined (GRUB_UTIL) && defined(ENABLE_NLS) && ENABLE_NLS)
   char *dst_locale;
 
   dst_locale = grub_util_path_concat (2, dst, "locale");
